content/news/2021/2021-09-01-announcing-the-new-exceptionless-javascript-client.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Announcing The New Exceptionless JavaScript Client
+date: 2021-09-03
+draft: false
+---
+    
+We have had a JavaScript client for several years, but it was quickly becoming out of date. To help address the issues with a fast-moving JavaScript landscape and to extend our JavaScript support, we have recently a completely rewritten client that supports multiple JavaScript frameworks. 
+
+If you just want to dive into the documentation, [you can review that here](../../docs/clients/javascript/index.md). If you want the story behind the rewrite, let's talk about some of the challenges we wanted to solve. 
+
+1. Support for additional frameworks
+2. Bring the client up to modern JavaScript standards 
+3. Have a little more code separation
+4. Provide a better developer experience
+
+We'll talk about each one of these points in a little more detail. 
+
+## Support for additional frameworks 
+
+When Exceptionless launched, the JavaScript ecosystem was powered by Bower, front-end applications were built with jQuery, and supporting IE6 was a legitimate concern on the minds of many developers. Over time, the ecosystem grew. We got React, Vue, Express, Typescript support became popular, and React Native made it easy for web devs to jump into building mobile apps. 
+
+While Exceptionless continued to work well for most use cases, it was clear we needed an updated client to support the exploding ecosystem. React Native was one of our specific targets because Exceptionless's JS client flat-out couldn't work in React Native. But while we were at it, we wanted to make Vue.js support easier, add some native React functionality, and more. What we arrived at is a more robust client with individual sub-clients for various frameworks. 
+
+## Bring the client up to modern JavaScript standards
+
+ECMAScript has progressed a lot in the last ten years. It was clear that we would need to re-write our client to keep up. With our new implementation, it will be much easier to stay on top of new features and experimental features. 
```

Comment:
We were already using typescript which kept the client somewhat modern. It was just our api surface couldn't be broken without a major bump, and without that we couldn't use modern features. 
We may want to talk about some of the new features here. 

- Like ESM solves the issues we had issues with module bundling. We had to have a special umd bundle which supported universal apps (browser and node) and this caused a bunch of issues.
-  We also had a lot of legacy JS usages (callbacks, and IE6 support) which prevented us from simplifying api usages for plugins. Was also harder to support Promises for things like getting source maps and modern submission methods like fetch. 
- We are also now using modern browser api's to pause the client when not in focus/active window so the page can go to sleep easier and thus longer battery life

Reply:
Also not sure if it's worth comparing code wise some of the old to the new usages, to show off how much easier it is.